conans/client/cache/editable.py

```diff
@@ -42,3 +42,6 @@ def remove(self, ref):
             self.save()
             return True
         return False
+
+    def update(self, workspace_edited):
+        self._edited_refs.update(workspace_edited)
```

Comment:
_update -vs- substitution._ 

I see no concerns right now, but I wanted to point this out and think about use cases.

Reply:
Still needed? I saw that you were doing `self._cache.editable_packages = workspace.get_editable_dict()` when creating the workspace.

----
IMO I prefer to keep the information of the editable packages being used inside this class (easier to debug), so I would vote for some code like:

_conan_api.py_
```python

    with self._cache.editable_packages.override(workspace.get_editable_dict()):

         recorder = ActionRecorder()
         deps_graph, _ = self._graph_manager.load_graph(workspace.root, None, graph_info, build,
                                                       False, update, remote_name, recorder)

         print_graph(deps_graph, self._user_io.out)

         installer = BinaryInstaller(self._cache, self._user_io.out, self._remote_manager,
                                    recorder=recorder, hook_manager=self._hook_manager)
         installer.install(deps_graph, keep_build=False, graph_info=graph_info)
         workspace.generate(cwd, deps_graph)
```

_editable.py_
```python
    ...
    @contextmanager
    def override(self, editable_packages):
        old = self._edited_refs
        try:
            self._edited_refs = editable_packages
            yield
        finally:
            self._edited_refs = old
```

Reply:
Fixing it. I won't use the context manager, because actually, the @api_method should manage to provide a clean environment of ``self._cache`` (already opened engineering issue)